src/CarbonAware.Tools/CarbonAware.Tools.WattTimeClient/src/WattTimeClient.cs

```diff
@@ -120,15 +120,14 @@ public async Task<IEnumerable<Forecast>> GetForecastByDateAsync(string balancing
 
         var result = await this.MakeRequestAsync(Paths.Forecast, parameters, tags);
 
-        var forecasts = JsonSerializer.Deserialize<List<Forecast>>(result, options) ?? throw new WattTimeClientException($"Error getting forecasts for  {balancingAuthorityAbbreviation}");
-
-        return forecasts;
+        var forecasts = JsonSerializer.Deserialize<List<Forecast>>(result, options) ?? throw new WattTimeClientException($"Error getting forecasts for {balancingAuthorityAbbreviation}");
+        return forecasts.FirstOrDefault();
```

Comment:
qq: if the Method always expects 1 valid result, is using `FirstOrDefault` truly ok? 
If not, I would recommend it either Log _or_ Throw that something unexpected was returned from WattTime...?

Reply:
We didn't want to through because it's not a WattTime error that the user requested a time when there was no forecast generated, that's a user error. So we decided to include the error checking one level up, at the data source level, which will then throw with the message that the user's requested at time didn't have a forecast

Reply:
@dnastrain , it is passing only one time (start=end from wartime perspective). So worst case scenario is that there is no 'sampling' at the provided time, therefore the array returned by wartime api, is going to be empty, then we send back a null.
If you pass a start != end to watttime, you would get more instances.

Reply:
true, I agree with you that's how watttime works today @JenMadiedo , and so I guess what I'm saying is that "if that ever changed, and wattime returned more than 1 forecast when we're expecting 1-and-only-1, we might want to at least log.warn about it" :)